plugins/org.eclipse.glsp.graph/src/org/eclipse/glsp/graph/builder/impl/GArguments.java

```diff
@@ -0,0 +1,83 @@
+/********************************************************************************
+ * Copyright (c) 2020 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.graph.builder.impl;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.eclipse.emf.common.util.EMap;
+
+public final class GArguments extends LinkedHashMap<String, Object> {
```

Comment:
In contrast to the `GLayoutOptions` the arguments map can hold arbitary key-value pairs so explicit subclassing and limiting the values you can put into the map doesn't make sense Imo.
I think this rather should be a `GArgumentsUtil` instead of Map subclass (similar to the util class for `ClientOptions`)

Reply:
I'm not sure I fully agree since the `GArguments` map can also hold arbitrary key-value pairs but It's better to be consistent ;-) I changed the structure to a utility class similar to `ClientOptions`.